homeassistant/components/aprs/device_tracker.py

```diff
@@ -0,0 +1,129 @@
+ATTR_FROM = 'from'
+ATTR_FORMAT = 'format'
+ATTR_POS_AMBIGUITY = 'posambiguity'
+ATTR_SPEED = 'speed'
+
+CONF_CALLSIGN = 'callsign'
+CONF_CALLSIGNS = 'callsigns'
+
+DEFAULT_CALLSIGN = 'N0CALL'
+DEFAULT_HOST = 'rotate.aprs2.net'
+
+MSG_FORMATS = ['compressed', 'uncompressed', 'mic-e']
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_CALLSIGNS): cv.ensure_list,
+    vol.Optional(CONF_CALLSIGN,
+                 default=DEFAULT_CALLSIGN): cv.string,
+    vol.Optional(CONF_HOST,
+                 default=DEFAULT_HOST): cv.string
+})
+
+
+def make_filter(callsigns: list) -> str:
+    """Make a server-side filter from a list of callsigns."""
+    return ' '.join('b/{0}'.format(cs.upper()) for cs in callsigns)
+
+
+def setup_scanner(hass, config, see, discovery_info=None):
+    """Set up the APRS tracker."""
+    callsigns = config.get(CONF_CALLSIGNS)
+    fltr = make_filter(callsigns)
+
+    callsign = config.get(CONF_CALLSIGN)
+    host = config.get(CONF_HOST)
+    aprs_listener = AprsListenerThread(
+        callsign, host, fltr, see)
+
+    def aprs_disconnect(event):
+        aprs_listener.stop()
+
+    def aprs_connect(event):
+        try:
+            aprs_listener.start()
+        except OSError:
+            _LOGGER.error("Failed to connect to %s with callsign %s",
+                          CONF_HOST, CONF_CALLSIGN)
+        finally:
+            hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP,
+                                 aprs_disconnect)
+
+    hass.bus.listen_once(EVENT_HOMEASSISTANT_START,
+                         aprs_connect)
+
+    return True
+
+
+class AprsListenerThread(threading.Thread):
+    """APRS message listener."""
+
+    def __init__(self, callsign, host, fltr, see):
+        """Initialize the class."""
+        super().__init__()
+
+        import aprslib
+
+        self.callsign = callsign
+        self.host = host
+        self.fltr = fltr
+        self.see = see
+        self.ais = aprslib.IS(self.callsign, host=self.host, port=14580)
+
+    def run(self):
+        """Connect to APRS and listen for data."""
+        self.ais.set_filter(self.fltr)
+        self.ais.connect()
+        try:
+            _LOGGER.info("Opening connection to %s with callsign %s",
+                         self.host, self.callsign)
+            self.ais.consumer(callback=self.rx_msg, immortal=True)
+        except OSError:
+            _LOGGER.info("Closing connection to %s with callsign %s",
+                         self.host, self.callsign)
+
+    def stop(self):
+        """Close the connection to the APRS network."""
+        self.ais.close()
+
+    def rx_msg(self, msg):
+        """Receive message and process if position."""
+        _LOGGER.debug("APRS message received: %s", str(msg))
+        if msg[ATTR_FORMAT] in MSG_FORMATS:
+            dev_id = slugify(msg[ATTR_FROM])
+            lat = msg[ATTR_LATITUDE]
+            lon = msg[ATTR_LONGITUDE]
+
+            attrs = {}
+            if ATTR_ALTITUDE in msg:
```

Comment:
You've accessed latitude and longitude in line 114 already. This check isn't needed.

Reply:
That's 'altitude', not 'latitude'. 🙂 I want to include it in the attributes if it's present.

Reply:
Ups! My fault.

Reply:
You could clean-up the code here if you introduce a for loop for all handled attributes.

Reply:
I originally had it that way but split them out because I wanted to change the names. But I removed the name changing so I'll change it back to a loop.